Source/JavaScriptCore/runtime/ImportMap.cpp

```diff
@@ -225,10 +224,40 @@ Expected<void, String> ImportMap::registerImportMap(const SourceCode& sourceCode
         return codePointCompareLessThan(rhs.m_scope.string(), lhs.m_scope.string());
     });
 
+    Integrity integrity;
+    if (auto integrityValue = rootMap->getValue("integrity"_s)) {
+        auto integrityMap = integrityValue->asObject();
+        if (!integrityMap)
+            return makeUnexpected("integrity is not a map"_s);
+
+        // https://html.spec.whatwg.org/C#normalizing-a-module-integrity-map
+        for (auto& [key, value] : *integrityMap) {
+            URL integrityURL = resolve(key, baseURL);
+            if (UNLIKELY(integrityURL.isNull())) {
+                // XXX - here we want to print out an console error without returning - "Integrity URL is not a valid URL starting with '/', './' or '../'"
```

Comment:
I'd love thoughts/pointers on the best way to do that

Reply:
If you can access the document, maybe:

```c++
document->addConsoleMessage(MessageSource::Security, MessageLevel::Warning, "Integrity URL is not a valid URL starting with '/', './' or '../'"_s);
```

Reply:
I don't think we can do that from inside JSCore. But I think I now have a better understanding of how to send those strings to bindings